src/intl-head.tsx

```diff
@@ -0,0 +1,28 @@
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import React, { ReactElement, ReactNode } from 'react';
+import { useAlternateLinks } from './hooks/useAlternateLinks';
+import { useCanonicalUrl } from './hooks/useCanonicalUrl';
+
+interface IntlHeadProps {
+  title: string;
+  children?: ReactNode;
+  currentLocale?: string;
```

Comment:
I don't understand the reason for this prop. Why not always use the prop provided by the router?

Reply:
Right now I've disabled automatic language detection and we are doing it in `gettingServerSideProsp` in index. As long as the user hasn't selected a locale with the language switcher, then the locale in router and the locale grabbed from the browser will be different. That's why I'm sending the locale. When we are navigating a page with a locale, then this will be the locale in the router and we wont need to pass it.